app/controllers/restaurants_controller.rb

```diff
@@ -1,16 +1,62 @@
+require_relative './application_controller'
+require 'active_record'
+
 class RestaurantsController < ApplicationController
-  def list
+  before_action :set_restaurant, only: [:show, :update, :destroy]
+  #TODO: consider adding authentication
+  rescue_from ActiveRecord::RecordNotFound, :with => :invalid_id_handler
+
+  # GET /restaurants
+  def index
+    @restaurants = Restaurant.all
+    @restaurants = @restaurants.where('cuisine = ?', query_params["by_cuisine"]) if query_params.has_key? "by_cuisine"
```

Comment:
Consider putting all these specific indices into separate endpoints in the controller, so something like:
`restaurants_by_cuisine`
`restaurants_by_rating` 
etc. etc.
Instead of handling the logic inside one specific endpoint. That way you don't need to do an "if", each endpoint is easily identifiable and its easy to understand what it does at a glance. Also if you do it that way the function can look like this:

```
def restaurants_by_cuisine
  @restaurants = @restaurants.where('cuisine = ?', params.require('cuisine'))
  render :json => @restaurants
end
```

Notice that I call params.require, since for this method do work I must have a query param which indicates which cuisine I want to search by

Reply:
But how would that work for a combination of conditions? (if somone wants an Italian restaurant with 3 stars and delivery time up to an hour)

Reply:
Oh I see where you're going with this. 
Most people do the filtration via the Frontend only, so they get a list of all restaurants from the backend, and then work on that list based on the users selection in the UI

(This isn't to say if its right or wrong, both are good :) )

If you want to filter it in the backend then yeah, you will need an endpoint where you can do all three. I still advise doing it in a different endpoint than index, go for `search` for example. Just so that `index` can be used to get all of the restaurants without expecting any query param at all